api/src/main/resources/api/service_openapi.yaml

```diff
@@ -20,6 +20,28 @@ paths:
         500:
           description: Service is broken
 
+  # TODO implement "real" export endpoints
+  '/underlays/{underlay}/entities/{entity}/dataset:generateSqlQuery':
+    parameters:
+      - $ref: '#/components/parameters/Underlay'
+      - $ref: '#/components/parameters/Entity'
+    post:
+      # TODO authorization
+      security: [ ]
+      summary: Returns an SQL query for selecting a dataset of an Entity's attributes.
+      operationId: generateSqlQuery
+      tags: [EntitiesDataset]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/EntityDataset'
```

Comment:
In this case, I would probably make a dedicated request schema for this method for extensibility. Usually I would just make the request/responses parallel, e.g. `GenerateSqlQueryRequest`

Reply:
Good idea, thanks. Went with `GenerateSqlQueryDatasetRequest` since we'll have a few amount of sql generations.